pandas/_libs/reshape.pyx

```diff
@@ -124,6 +126,9 @@ def explode(ndarray[object] values):
     counts = np.zeros(n, dtype='int64')
     for i in range(n):
         v = values[i]
+        if isinstance(v, abc.Set):
```

Comment:
In this case, I think we can just use `if c_is_list_like(v, True):`

Reply:
Yes, that seems better. Regarding the comment about always casting to list, wouldn't we want to avoid unnecessary casting if possible?

Reply:
I guess it's always casting to list vs always calling `isinstance`

You could mircobenchmark to see if there's a significant difference between the options.

Reply:
You're right, always casting looks like it's more efficient. Kind of interesting it does better even when the cast is "redundant" (I guess the isinstance check is actually pretty expensive):
```python
n = 10_000
ser = pd.Series(n * [list(range(50))])
%timeit ser.explode()
# with isinstance
# 20.3 ms ± 383 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
# always cast
# 18.2 ms ± 697 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```